modules/dynamic-plugins/proc-configuring-events-module-for-github.adoc

```diff
@@ -0,0 +1,117 @@
+
+Learn how to configure Events Module for use with the {product-very-short} GitHub Discovery feature,
+and GitHub organization data.
+This is a Technology Preview feature.
+
+[IMPORTANT]
+====
+Technology Preview features provide early access to upcoming product innovations, enabling you to test functionality and provide feedback during the development process.
+However, these features are not fully supported under Red Hat Subscription Level Agreements, may not be functionally complete, and are not intended for production use.
+As Red Hat considers making future iterations of Technology Preview features generally available, we will attempt to resolve any issues that customers experience when using these features.
+See: link:https://access.redhat.com/support/offerings/techpreview/[Technology Preview support scope].
+====
+
+.Prerequisites
+* For GitHub Discovery only: you have enabled {product-docs-link}/html-single/integrating_red_hat_developer_hub_with_github/index#enabling-github-repository-discovery[GitHub Discovery].
+
+.Procedure
+. Update your `dynamic-plugins.yaml` file with GitHub Discovery or GitHub organization data.
+.. For GitHub Discovery, use the following file example as a guideline:
++
+[source,yaml]
+----
+data:
+dynamic-plugins.yaml: |
+includes:
+- dynamic-plugins.default.yaml
+plugins:
+- package: ./dynamic-plugins/dist/backstage-plugin-catalog-backend-module-github-dynamic
+disabled: false
+pluginConfig:
+catalog:
+providers:
+github:
+providerId:
+organization: <your_organization_name>
+schedule:
+frequency: { hours: 24 }
+timeout: { minutes: 3 }
+----
+.. For GitHub organization data, use the following file example as a guideline:
++
+[source,yaml]
+----
+data:
+dynamic-plugins.yaml: |
+includes:
+- dynamic-plugins.default.yaml
+plugins:
+- package: ./dynamic-plugins/dist/backstage-plugin-catalog-backend-module-github-org-dynamic
+disabled: false
+pluginConfig:
+catalog:
+providers:
+githubOrg:
+id: production
+githubUrl: https://github.com
+orgs: [ '<your_organization_name>' ]
+schedule:
+frequency:
+hours: 24
+initialDelay:
+seconds: 15
+timeout:
+minutes: 10
+----
+. Add your GitHub integration credentials in the `app-config.yaml` file:
++
+[source,yaml]
+----
+integrations:
+github:
+- host: github.com
+token: ${GITHUB_PAT_TOKEN}
+----
+. Create a GitHub webhook with the following specifications:
+** *Events*: push, repository
+** *Content Type*: application/json
+** *Payload URL*: https://<your_instance_name>/api/events/http/github
+
+. Add the GitHub Events Module to your `dynamic-plugins.yaml` configuration file:
++
+[source,yaml]
+----
+data:
+dynamic-plugins.yaml: |
+includes:
+- dynamic-plugins.default.yaml
+plugins:
+- package: oci://04kash/events-backend-module-github:v0.0.1!backstage-plugin-events-backend-module-github
+disabled: false
+----
+
+. Add the following HHTP events configuration to the `app-config.yaml` file:
++
+[source,yaml]
+----
+events:
+http:
+topics:
+- github
```

Comment:
Fix indentation:
```suggestion
events:
  http:
    topics:
      - github
```


I think we should also mention adding the `webhookSecret` as it's done [upstream](https://backstage.io/docs/integrations/github/discovery/#install-and-configure-github-events-module) to ensure it's set up securely:

```suggestion
events:
  modules:
    github:
      webhookSecret: ${GITHUB_WEBHOOK_SECRET}
```